File: framework/languages/java/samples/java/GameQuest/Server/GameApi/src/main/java/systems/zlink/samples/gamequest/server/gameapi/handlers/GetGameplaySnapshotHandler.java
package systems.zlink.samples.gamequest.server.gameapi.handlers;

import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.gamequest.server.gameapi.store.GameQuestStore;
import systems.zlink.samples.gamequest.shared.contracts.Messages;

@ZLinkHandlerGroup("gameapi")
public final class GetGameplaySnapshotHandler
    implements ZLinkRequestHandler<Messages.GetGameplaySnapshotReq, Messages.GetGameplaySnapshotRes> {
    private final GameQuestStore store;

    public GetGameplaySnapshotHandler(GameQuestStore store) {
        this.store = store;
    }

    @Override
    public Messages.GetGameplaySnapshotRes handle(
        Messages.GetGameplaySnapshotReq request, ZLinkRequestContext context) {
        return store.readSnapshot(request.playerId());
    }
}

File: framework/languages/java/samples/kotlin/GameQuest/Server/GameApi/src/main/kotlin/systems/zlink/samples/kotlin/gamequest/server/gameapi/handlers/GetGameplaySnapshotHandler.kt
package systems.zlink.samples.kotlin.gamequest.server.gameapi.handlers

import systems.zlink.framework.channels.ZLinkRequestContext
import systems.zlink.framework.kotlin.ZLinkSuspendingRequestHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.samples.kotlin.gamequest.server.gameapi.infrastructure.store.GameQuestStore
import systems.zlink.samples.kotlin.gamequest.shared.contracts.GetGameplaySnapshotReq
import systems.zlink.samples.kotlin.gamequest.shared.contracts.GetGameplaySnapshotRes

@ZLinkHandlerGroup("gameapi")
class GetGameplaySnapshotHandler(
    private val store: GameQuestStore,
) : ZLinkSuspendingRequestHandler<GetGameplaySnapshotReq, GetGameplaySnapshotRes> {
    override suspend fun handle(request: GetGameplaySnapshotReq, context: ZLinkRequestContext): GetGameplaySnapshotRes =
        store.readSnapshot(request.playerId)
}
